pkg/loader/tf.go

```diff
@@ -1057,3 +1064,23 @@ func rangeToLocation(r hcl.Range) Location {
 		Col:  r.Start.Column,
 	}
 }
+
+// TfFilePathJoin is like `filepath.Join` but avoids cleaning the path.  This
+// allows to get unique paths for submodules including a parent module, e.g.:
+//
+//     .
+//     examples/mssql/../../
+//     examples/complete/../../
+//
+func TfFilePathJoin(leading, trailing string) string {
+	if filepath.IsAbs(trailing) {
+		return trailing
+	} else if leading == "." {
+		return trailing
+	} else {
+		sep := string(filepath.Separator)
```

Comment:
```suggestion
		trailing = filepath.FromSlash(trailing)
		sep := string(filepath.Separator)
```

This can come out kind of strange on Windows when the path associated with the module uses the unix path separator:

```
 [2]: module.rds_instance.aws_db_instance.default
       in terraform-aws-rds\examples\complete\../../\main.tf:22:1
       included at terraform-aws-rds\examples\complete\main.tf:29:26
```

`FromSlash` ensures that they have the appropriate path separator. I think that Windows path separators are invalid in the module source field, so I don't think we have to worry about the same in reverse.